Source/JavaScriptCore/runtime/JSArrayBufferConstructor.cpp

```diff
@@ -96,8 +96,10 @@ EncodedJSValue JSGenericArrayBufferConstructor<sharingMode>::constructImpl(JSGlo
     if (!buffer)
         return JSValue::encode(throwOutOfMemoryError(globalObject, scope));
     
-    if (sharingMode == ArrayBufferSharingMode::Shared)
+    if (sharingMode == ArrayBufferSharingMode::Shared) {
         buffer->makeShared();
+        ASSERT(!buffer->isDetached());
```

Comment:
Can this ASSERT be in `ArrayBuffer::makeShared()` instead?

Reply:
Sounds good, done.